controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller_intg_test.go

```diff
@@ -88,6 +88,16 @@ func virtualMachinePublishRequestReconcile() {
 			Expect(ctx.Client.Status().Update(ctx, vmObj)).To(Succeed())
 
 			Expect(ctx.Client.Create(ctx, cl)).To(Succeed())
+
+			// Ensure VM and CL all exist before VMPub is created.
+			Eventually(func() error {
```

Comment:
By the time the pub req is first reconciled there is no guarantee that either of those still exists. Or another race to consider is that since the webhooks run on all replicas - not just the leader that runs the controllers - is that what one replica currently thinks does or does not exists may not agree with another replica.

Now ctrl-runtime Manager does have a APIReader client too that by-passes the cache but that still doesn't mean that whatever is checked at the webhook is still true on the reconcile. I think a valid use case too - and that fits in the k8s model - is to create the VM and pub req at the "same time" so the pub req gets processed when the VM is sufficiently created.

Reply:
I don't want this webhook change to block this default item name fix. Will merge this first and refactor webhook later. This is another story.